src/pudl/dbt_wrapper.py

```diff
@@ -75,14 +122,27 @@ def build_with_context(
     dbt = dbtRunner()
     dbt_dir = PUDL_ROOT_PATH / "dbt"
 
-    # 2025-06-06 TODO: use the --project-dir flag instead of chdiring all the time?
     with chdir(dbt_dir):
-        _ = dbt.invoke(["deps"])
-        _ = dbt.invoke(["seed"])
+        dbt.invoke(["deps"])
+        dbt.invoke(["seed"])
         build_output: dbtRunnerResult = dbt.invoke(["build"] + cli_args)
         build_results = cast(RunExecutionResult, build_output.result)
 
-    failed_nodes = get_failed_nodes(build_results)
-    failure_contexts = get_context_for_nodes(failed_nodes)
+    failed_nodes = __get_failed_nodes(build_results)
+
+    weighted_quantile_failures, compiled_sql_failures = [], []
```

Comment:
Decided to do the splitting at the caller here instead of deeper in the code so that each function can have a simpler spec.

We could have a dictionary of handlers instead of hard-coding the handlers, but with two I think variables is nicer than dictionary keys.